statsmodels/discrete/discrete_model.py

```diff
@@ -446,14 +450,16 @@ def _derivative_exog_helper(self, margeff, params, exog, dummy_idx,
 
 class BinaryModel(DiscreteModel):
 
-    def __init__(self, endog, exog, **kwargs):
-        super(BinaryModel, self).__init__(endog, exog, **kwargs)
+    def __init__(self, endog, exog, offset=None, **kwargs):
+        super(BinaryModel, self).__init__(endog, exog, offset=offset, **kwargs)
         if (not issubclass(self.__class__, MultinomialModel) and
                 not np.all((self.endog >= 0) & (self.endog <= 1))):
             raise ValueError("endog must be in the unit interval.")
+        if offset is None:
+            delattr(self, 'offset')
```

Comment:
Is this necessary?  Why not keep `self.offset = None`?  Then do `if self.offset is not None` checks instead of `if hasattr(self, 'offset')`.